examples/2-multi-step.ipynb

```diff
@@ -0,0 +1,411 @@
+    "                 cache_dirpath='./cache')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pca_step = Step(name='PCA',\n",
+    "                transformer=PCATransformer(),\n",
+    "                input_steps=[norm_step],                 \n",
+    "                cache_dirpath='./cache')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Our classifier step will have to combine two data flows: the features processed by PCA, and the labels fed directly from input. Therefore, we will have to use the `adapter` argument to specify how to map those inputs to transformer arguments."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rf_step = Step(name='LogReg',\n",
+    "               transformer=LogRegTransformer(),\n",
+    "               input_steps=[pca_step],\n",
+    "               input_data=['input'],           # TODO: Why string for input_data but Python identifier for input_steps?\n",
+    "               adapter={'X':([('PCA', 'X')]),  # TODO: Why string here but identifier when specifying input steps?\n",
+    "                        'y':([('input', 'y')]),\n",
+    "                        },\n",
+    "               cache_dirpath='./cache')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "One may think it's a bit cumbersome to create your transformers and then have to wrap them with steps. However, there is an advantage to this - think about it:\n",
+    "* The **transformer** is the ***implementation*** of a machine learning algorithm. It has an input and outputs but it doesn't even know what these are connected to.\n",
+    "* The **steps** define the ***connections*** between different transformers. At this level of abstraction, all the algorithmic details are hidden. The code that defines steps and connects them together is compact and it's easier to see what is connected to what."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "So what does our pipeline look like?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rf_step"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This looks about right - let's move on to training!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Training\n",
+    "\n",
+    "Training a pipeline is a one-liner:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "preds_train = rf_step.fit_transform(data_train);  # TODO: It's weird that we're fitting the final step. I get the idea but it still seems unintuitive that the last step represents the whole pipeline..."
```

Comment:
@mromaniukcdl fair point. We should discuss it at the next planning.

Reply:
@mromaniukcdl @jakubczakon I think it is just a matter of explanation that you run `pipeline(data)`